update chronological-updates for ganache use with wsl (#1440)
smartcontractkit/full-blockchain-solidity-course-js: chronological-updates.md

# Chronological Updates

This file will contain all updates to sections.

# Lesson 5

## Windows, WSL, & Ganache

- Per [this question](https://github.com/smartcontractkit/full-blockchain-solidity-course-js/discussions/34#discussioncomment-2846436), if you're using WSL, for the ganache UI you'll have to use a different endpoint.

You have 4 options to fix this:

1. Use the WSL endpoint on the ganache UI
2. Use `ganache` from the command line
3. Use `ganache-cli` from the command line (not recommended)
4. Just use `hardhat` from the command line (you'll have to use hardhat at some point anyways!)

### Using the WSL Endpoint

On the Ganache UI, you can select a different hostname and connect to whatever IP you see. In the example below, you'd connect to `172.29.0.1`

![img](./img/ganache-windows.png)

You can also try to edit it by:

1. Inside Ganache in the top-right corner tap on the gear icon (settings).
2. Then in the server tab, change the HOSTNAME and select `0.0.0.0 - All Interfaces`.

> 3. Now replace your RPC Server URL inside the code with this latest URL that will be something like: http://172.29.0.1:7545. Recheck this by running this command:

Reviewer: ```suggestion
3. Now replace the RPC Server URL (inside the code) with this updated URL which will be something like `http://172.29.0.1:7545`. Verify this by running this command:
```

   `grep -m 1 nameserver /etc/resolv.conf | awk '{print $2}'`.

> **IF it gets stuck** at `"Deploying, please wait..."`, then you have to allow the connection for Ganache in `Windows Defender Firewall with Advanced Security`.

Reviewer: ```suggestion
*If it gets stuck* at `"Deploying, please wait..."`, then you have to allow the connection for Ganache in `Windows Defender Firewall with Advanced Security`.
```

Italics are more apt (by convention) in this context.
